system/core/language.php

```diff
@@ -24,9 +24,16 @@ function __($msgid, $args = null) {
 	*/
 	function languageOptionList($select, $omit) {
 		global $isoLangList;
-		$html = '';
-		foreach($isoLangList as $code => $langName) if ($code!=$omit) $html.= '<option value='.$code.($code==$select ? ' selected' : '').'>'.$code.': '.$langName.'</option>';
-		return addslashes($html);
+		$langList = hypha_getUsedContentLanguages();
+
+		$inUse = '';
+		foreach ($langList as $code) if ($code!=$omit) $inUse.= '<option value='.$code.($code==$select ? ' selected' : '').'>'.$code.': '.$isoLangList[$code].'</option>';
+		$html = '<optgroup label="in use">' . addslashes($inUse) . '</optgroup>';
```

Comment:
This is probably a good moment to add html escaping, and remove this pointless addslashes (probably in a subsequent commit, though).

Reply:
Tested it, and removing the `addslashes` results in errors.

Reply:
I've removed the addSlashes and added htmlspecialchars, just force pushed this.